File: api/src/main/java/com/comunica/news/models/Post.java
package com.comunica.news.models;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Builder
@Table(name = "tab_posts")
public class Post {

    @Id
    private String id;

    @Column(name = "titulo")
    private String titulo;

    @Column(name = "descricao")
    private String descricao;

    @Column(name = "img")
    private String img;

    @Column(name = "data")
    private LocalDate data;

    @Column(name = "autor")
    private String autor;

    @JoinColumn(name = "usuario_id", nullable = false)
    @ManyToOne
    private Usuario usuario;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "post")
    private List<Comentario> comentarios;
}

File: ComunicaNews/app/src/main/java/com/toddy/comunicanews/models/Post.kt
package com.toddy.comunicanews.models

data class Post(
    var id: String? = "",
    var titulo: String? = "",
    var descricao: String? = "",
    var autor: String? = "",
    var data: String? = "",
    var likes: Int? = 0,
    var comentarios: List<Comentario> = listOf()
)